cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java

```diff
@@ -200,8 +202,9 @@ public void drop(final DatasetId datasetInstanceId, final DatasetTypeMeta typeMe
     LOG.info("Dropping dataset with spec: {}, type meta: {}", spec, typeMeta);
     try (DatasetClassLoaderProvider classLoaderProvider =
            new DirectoryClassLoaderProvider(cConf, locationFactory)) {
+      UserGroupInformation ugi = getUgiForDataset(impersonator, datasetInstanceId);
 
-      impersonator.doAs(datasetInstanceId.getParent(), new Callable<Void>() {
+      ImpersonationUtils.doAs(ugi, new Callable<Void>() {
```

Comment:
We don't want to lookup owners for system dataset because it will be null, so we use the UserGroupInformation for the current user.